main.py

```diff
@@ -57,14 +57,14 @@ async def async_main(timestamp):
 
     existing_titles = get_all_titles(articles_sheet)
     providers = get_all_providers(providers_sheet)
-    fetcher = PageFetcher()
+    fetcher_state = init_fetcher_state()
```

Comment:
i think there should be a space between fetcher_state and providers, so it is more clear